jupyterhub/templates/proxy/autohttps/deployment.yaml

```diff
@@ -41,90 +29,66 @@ spec:
       terminationGracePeriodSeconds: 60
       nodeSelector: {{ toJson .Values.proxy.nodeSelector }}
       {{- include "jupyterhub.coreAffinity" . | nindent 6 }}
+      volumes:
+        - name: certificates
+          emptyDir: {}
+        - name: traefik-config
+          configMap:
+            name: traefik-proxy-config
+      initContainers:
+      - name: volume-mount-hack-why-god-still
```

Comment:
I have some ideas to avoid the mount hack! Perhaps read the Woops section first below though.

## 1 - Read [here under "Secret files permissions"](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets)
Perhaps we use volumes `defaultMode` or volumeMounts `mode` configuration? Note that working in YAML, you can use octal specification even though you can't in JSON to specify.

## 2 - Mounting a specific key/value pair
I have two reasons why I figure this may be worth trying.

1. If we mount a specific key/value pair as a file, rather than the whole directory, there may be a significant difference for how the permissions are applied. I figure they may apply on the root of the mount rather than the recursively, leaving me uncertain about what will happen in general.

2. It sais in the official k8s documentation "A container using a Secret as a subPath volume mount will not receive Secret updates.". It probably is a good idea to not have that update mechanism active while having the sync-script do work at the same time.

## Example solution to avoid mount hack

```yaml
volumes:
  - name: certificates
    emptyDir: {}
    defaultMode: 0755
containers:
  volumeMounts:
    - name: certificates
      # Here we specify a file rather than a directory, as we specify a subPath
      # below to select the mounting of a single key/value pair in the secret
      mountPath: /etc/acme/acme.json
      subPath: acme.json
```

## Woops...

I forgot that you do not interact with a k8s secret volume here but a emptyDir volume, you do it all using kubernetes API from the sync-script... Hehehe. Ah well, but perhaps some points were of relevance still so I'll leave the text above unmodified.

Reply:
Great catch, @consideRatio. I have removed the hack with a defaultMode instead.

Reply:
Correction - emptyDir defaults to the perms we want anyway, so I just removed the hack and defaultDir :)